packages/@aws-cdk/aws-iam/lib/policy-statement.ts

```diff
@@ -0,0 +1,313 @@
+import cdk = require('@aws-cdk/cdk');
+import { AccountPrincipal, AccountRootPrincipal, Anyone, ArnPrincipal, CanonicalUserPrincipal,
+  FederatedPrincipal, IPrincipal, ServicePrincipal, ServicePrincipalOpts } from './principals';
+import { mergePrincipal } from './util';
+
+/**
+ * An abstract PolicyStatement
+ */
+export interface IPolicyStatement {
+  /**
+   * Render the policy statement to JSON
+   */
+  toStatementJson(): any;
+}
+
+/**
+ * Represents a statement in an IAM policy document.
+ */
+export class PolicyStatement implements IPolicyStatement {
```

Comment:
Should we make a linter rule to enforce the usage of IPolicyStatement everywhere?